library/lms.c

```diff
@@ -499,13 +498,20 @@ static int get_merkle_path( mbedtls_lms_private_t *ctx,
                             unsigned int leaf_node_id,
                             unsigned char *path )
 {
-    unsigned char tree[MERKLE_TREE_NODE_AM_MAX][MBEDTLS_LMS_M_NODE_BYTES_MAX];
     unsigned int curr_node_id = leaf_node_id;
     unsigned int adjacent_node_id;
     unsigned int height;
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    unsigned char *tree = NULL;
 
-    ret = calculate_merkle_tree( ctx, ( unsigned char * )tree );
+    tree = mbedtls_calloc( MERKLE_TREE_NODE_AM(ctx->params.type),
+        MBEDTLS_LMS_M_NODE_BYTES(ctx->params.type) );
```

Comment:
Partly preexisting: defining a short name (`const size_t …`) for `MBEDTLS_LMS_M_NODE_BYTES( ctx->params.type )`, and perhaps also `MERKLE_TREE_NODE_AM( ctx->params.type )`, would make the code easier to read.

Also goes in `mbedtls_lms_calculate_public_key`.

Reply:
Done.